lib/lruhash.rb

```diff
@@ -1,279 +1,38 @@
-# A hash class that closely mimicks the original Ruby hash
-# with a subtle difference, it is limited in size and cannot
-# grow beyond a max value. This is done by maintaining the
-# hashed objects in activity order and removing the least
-# recently used item if the hash exceeds its size limit.
-# Useful for implementing in-process caches where memory
-# growth control is desired
-# 
-# There are two alternate implementations, one for Ruby 1.9.x+
-# and one for previous Ruby versions. The 1.9.x implementation
-# makes use of the fact that the new Ruby hash maintains insertion
-# order. It should be better than the version for older Rubys as it relies
-# on the native hash for most of its operations
-if RUBY_VERSION.to_f >= 1.9 
-  module LRU
-    def max
-      @max ||= 256
-    end
-
-    def max=(value)
-      @max = value
-      shift while self.length > @max
-    end
-      
-    def [](key)
-      if value = super(key)
-        self.delete(key)
-        self[key] = value
-      end
-    end
-    
-    def []=(key,value)
-      unless self[key]
-        self.shift while self.length > (max - 1)
-      end
-      super(key,value)
-    end
-  end
-
-  class LRUHash < Hash
-    include LRU
-  end
-
-else
+class LRUHash < Hash
+  def max
+    @max ||= 256

```

Comment:
@primiti  I think max ought to be a keyword argument to the constructor.  I'm not sure we need a default even.